### PR TITLE
Removing the check about the size re: ray-project/ray#3450

### LINKED ISSUE #3450: Allow greater than memory allocation for plasma store on Mac
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: OSX 10.12
- **Ray installed from (source or binary)**: source
- **Python version**: 3.6
- **Exact command to reproduce**: (See below)

### Describe the problem

In Modin, it would be great if we could specify `plasma_directory="/tmp"` and `object_store_memory=n * physical_mem`. It seems to work fine on Ubuntu, but I am getting the following error on Mac:

```Exception: The requested object store memory size is greater than the total available memory.```

### Source code / logs
```python
from psutil import virtual_memory
mem_bytes = virtual_memory().total
object_store_memory = 8 * mem_bytes
plasma_directory = "/tmp"
ray.init(
    redirect_output=True,
    include_webui=False,
    redirect_worker_output=True,
    use_raylet=True,
    ignore_reinit_error=True,
    plasma_directory=plasma_directory,
    object_store_memory=object_store_memory,
)
```

### PR DESCRIPTION
**Pull request by devin-petersohn**
I have opened this in the interest of continuing the discussion here.

## What do these changes do?

Removes the check causing #3450 

This is important in Modin specifically to allow us to use the disk for plasma.

## Related issue number
Resolves #3450

### CODE REVIEW

**Comment by robertnishihara** [comment]
Please don't merge this. If this happens it is almost always an error.

If we want to allow this then we need to make it's only happening in the case where the user passed in the object store memory and we need to provide a very scary warning.

**Comment by devin-petersohn** [comment]
I guess we should decide whether users should be responsible for the results of specifying `object_store_memory`. Was this causing some bugs previously?

I do not think raising a scary warning would be helpful to Modin. Printing the same information that was previously printed would be good enough for our purposes.

**Comment by pcmoritz** [comment]
I think the right thing to do here is to compare the object_store_memory with the size of the plasma_directory (for /tmp it would be the size of the disk, for /dev/shm it would be the size of the shared memory pool)

**Comment by devin-petersohn** [comment]
Updated PR given previous comments. 

It is not possible to verify the size of the device that is mounted for the `plasma_directory` in the general case. 

I moved the check inside the condition `if plasma_directory is None`.

File: python/ray/services.py
Comment (robertnishihara) [inline python/ray/services.py]: Why remove this?

Comment (robertnishihara) [reply python/ray/services.py]: Ok, I see you just moved it.

**Review by robertnishihara — APPROVED** [review]
This looks good to me assuming tests pass.

**Review by pcmoritz — APPROVED** [review]
fixed lint

**Comment by pcmoritz** [comment]
jenkins retest this please

**Pull request merged**
